src/librustc_privacy/lib.rs

```diff
@@ -911,7 +911,13 @@ impl<'a, 'tcx: 'a> SearchInterfaceForPrivateItemsVisitor<'a, 'tcx> {
     }
 
     fn predicates(&mut self) -> &mut Self {
-        self.tcx.predicates_of(self.item_def_id).visit_with(self);
+        let predicates = self.tcx.predicates_of(self.item_def_id);
+        for predicate in &predicates.predicates {
+            if let &ty::Predicate::Trait(/* Trait(poly_predicate)*/poly_predicate) = predicate {
```

Comment:
Hmm there's also a `Predicate::Projection` which used to reach `visit_trait_ref`.